src/main/java/org/zalando/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -120,7 +120,8 @@ public Properties getKafkaProducerProperties() {
         producerProps.put(ProducerConfig.COMPRESSION_TYPE_CONFIG, "lz4");
         producerProps.put(ProducerConfig.MAX_REQUEST_SIZE_CONFIG, kafkaSettings.getMaxRequestSize());
         producerProps.put(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, kafkaSettings.getDeliveryTimeoutMs());
-        producerProps.put(ProducerConfig.RETRIES_CONFIG, 0);
+        producerProps.put(ProducerConfig.RETRIES_CONFIG, 3);
+        producerProps.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, 1);
```

Comment:
Why?
As far as I know, we have 1 producer. 1 request per connection is a strange thing.

Reply:
Also, I see in configuration that this property is not prerequisite for retries.

Reply:
this is due to `retries` configuration:
```
Allowing retries without setting max.in.flight.requests.per.connection to 1 will potentially change the ordering of records because if two batches are sent to a single partition, and the first fails and is retried but the second succeeds, then the records in the second batch may appear first.
```
https://kafka.apache.org/documentation/#producerconfigs

Reply:
Yes, but for us it is working on a different level, cause we have many nakadi instances, so for us anyways real value = max.in.flight.requests.per.connection * amount of nakadi instances